test/unit/connection_string.test.js

```diff
@@ -224,81 +225,96 @@ describe('Connection String', function () {
-                      expect(options).to.have.property('loadBalanced', testOptions.loadBalanced);
-                    }
-                    if (
-                      test[1].parsed_options &&
-                      test[1].parsed_options.user &&
-                      test[1].parsed_options.password
-                    ) {
-                      expect(options.credentials.username).to.equal(test[1].parsed_options.user);
-                      expect(options.credentials.password).to.equal(
-                        test[1].parsed_options.password
-                      );
-                    }
-                  }
-                  done();
-                });
+                options = parseOptions(test.uri);
+                hosts = await promisify(resolveSRVRecord)(options);
               } catch (error) {
-                if (test[1].error) {
-                  expect(error).to.exist;
-                  done();
-                } else {
-                  throw error;
-                }
+                thrownError = error;
+              }
+
+              if (test.error) {
+                expect(thrownError).to.exist;
+                expect(hosts).to.not.exist;
+                return; // Nothing more to test...
+              }
+
+              expect(thrownError).to.not.exist;
+              expect(options).to.exist;
+
+              // Implicit SRV options must be set.
+              expect(options.directConnection).to.be.false;
+              const testOptions = test.options;
+              if (testOptions && 'tls' in testOptions) {
+                expect(options).to.have.property('tls', testOptions.tls);
+              } else if (testOptions && 'ssl' in testOptions) {
+                expect(options).to.have.property('tls', testOptions.ssl);
+              } else {
+                expect(options.tls).to.be.true;
+              }
+              if (testOptions && testOptions.replicaSet) {
+                expect(options).to.have.property('replicaSet', testOptions.replicaSet);
               }
+              if (testOptions && testOptions.authSource) {
+                expect(options).to.have.property('credentials');
+                expect(options.credentials.source).to.equal(testOptions.authSource);
+              }
+              if (testOptions && testOptions.loadBalanced) {
+                expect(options).to.have.property('loadBalanced', testOptions.loadBalanced);
+              }
+              if (test.parsed_options && test.parsed_options.user && test.parsed_options.password) {
+                expect(options.credentials.username).to.equal(test.parsed_options.user);
+                expect(options.credentials.password).to.equal(test.parsed_options.password);
+              }
+
+              // TODO(): srvMaxHost limiting happens in the callback given to resolveSRVRecord inside the driver
+              // How can we test this in unit test?
+
+              // if (options.srvHost && comment.includes('srvMaxHosts')) {
```

Comment:
Just a note but it's potentially one way to add a check... Could be that we test that the list of hosts after the randomisation is equal to the list of server description host names in the created topology? Just thinking aloud.

Reply:
Moved these tests to [initial_dns_seedlist_discovery.spec.test.ts](https://github.com/mongodb/node-mongodb-native/blob/NODE-3467/limit-mongos/test/integration/initial-dns-seedlist-discovery/initial_dns_seedlist_discovery.spec.test.ts#L20) to go along with our nice naming convention but also skipped them due to needing more concrete assertions around the error cases. And it is tracked here: NODE-3757